### ISSUE TITLE
Framework assertion in flutter/src/rendering/box.dart when using LayoutBuilder

### ISSUE THREAD

**Issue by llucax** [flutter/flutter#73676]
I'm reporting this because is what the error is suggesting:
```
I/flutter (23621): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (23621): more information in this error message to help you determine and fix the underlying cause.
I/flutter (23621): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (23621):   https://github.com/flutter/flutter/issues/new?template=2_bug.md
```

I tried this in Android and Web, and in Web it works without issues (with `flutter run -d chrome`). **Only Android is affected** for me. I'm using **Flutter beta** channel.

## Steps to Reproduce

1. Run `flutter create bug11`.
2. 
2. Update the files as follows (based on the example in https://flutter.dev/docs/cookbook/forms/validation#interactive-example):

**I marked important additions with `// IMPORTANT`, if any of these are removed, the error goes away.**

<details>
<summary>code sample</summary>

```dart
import 'package:flutter/material.dart';

void main() => runApp(MyApp());

class MyApp extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    final appTitle = 'Form Validation Demo';

    return MaterialApp(
      title: appTitle,
      home: Scaffold(
        appBar: AppBar(
          title: Text(appTitle),
        ),
        body: MyCustomForm(),
      ),
    );
  }
}

class Screen2 extends StatelessWidget {
  @override
  Widget build(BuildContext context) {
    return Scaffold(
      appBar: AppBar(
        title: const Text('Screen2'),
      ),
      body: Text('Success'),
    );
  }
}

// Create a Form widget.
class MyCustomForm extends StatefulWidget {
  @override
  MyCustomFormState createState() {
    return MyCustomFormState();
  }
}

// Create a corresponding State class.
// This class holds data related to the form.
class MyCustomFormState extends State<MyCustomForm> {
  // Create a global key that uniquely identifies the Form widget
  // and allows validation of the form.
  //
  // Note: This is a GlobalKey<FormState>,
  // not a GlobalKey<MyCustomFormState>.
  final _formKey = GlobalKey<FormState>();

  final _fieldFocusNode = FocusNode();

  bool _submitting = false;

  @override
  void dispose() {
    _fieldFocusNode.dispose();
    super.dispose();
  }

  void _submit() async {
    setState(() => _submitting = true);

    // Validate returns true if the form is valid, or false
    // otherwise.
    await Navigator.push<Screen2>(
        context, MaterialPageRoute(builder: (context) => Screen2()));

    setState(() => _submitting = false); // IMPORTANT
    _fieldFocusNode.requestFocus(); // IMPORTANT
  }

  @override
  Widget build(BuildContext context) {
    // Build a Form widget using the _formKey created above.
    return Form(
      key: _formKey,
      // IMPORTANT
      child: LayoutBuilder(
        builder: (context, constraints) => Column(
          crossAxisAlignment: CrossAxisAlignment.start,
          children: <Widget>[
            TextFormField(
              focusNode: _fieldFocusNode, // IMPORTANT
              readOnly: _submitting, // IMPORTANT
              onEditingComplete: _submit,
            ),
            Padding(
              padding: const EdgeInsets.symmetric(vertical: 16.0),
              child: ElevatedButton(
                onPressed: _submit,
                child: Text('Submit'),
              ),
            ),
          ],
        ),
      ),
    );
  }
}
```
</details>

<details><summary>Diff with the example code</summary>

```diff
diff --git a/lib/main.dart b/lib/main.dart
index a9ab2de..39f34c8 100644
--- a/lib/main.dart
+++ b/lib/main.dart
@@ -19,6 +19,18 @@ class MyApp extends StatelessWidget {
   }
 }
 
+class Screen2 extends StatelessWidget {
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      appBar: AppBar(
+        title: const Text('Screen2'),
+      ),
+      body: Text('Success'),
+    );
+  }
+}
+
 // Create a Form widget.
 class MyCustomForm extends StatefulWidget {
   @override
@@ -37,38 +49,52 @@ class MyCustomFormState extends State<MyCustomForm> {
   // not a GlobalKey<MyCustomFormState>.
   final _formKey = GlobalKey<FormState>();
 
+  final _fieldFocusNode = FocusNode();
+
+  bool _submitting = false;
+
+  @override
+  void dispose() {
+    _fieldFocusNode.dispose();
+    super.dispose();
+  }
+
+  void _submit() async {
+    setState(() => _submitting = true);
+
+    // Validate returns true if the form is valid, or false
+    // otherwise.
+    await Navigator.push<Screen2>(
+        context, MaterialPageRoute(builder: (context) => Screen2()));
+
+    setState(() => _submitting = false); // IMPORTANT
+    _fieldFocusNode.requestFocus(); // IMPORTANT
+  }
+
   @override
   Widget build(BuildContext context) {
     // Build a Form widget using the _formKey created above.
     return Form(
       key: _formKey,
-      child: Column(
-        crossAxisAlignment: CrossAxisAlignment.start,
-        children: <Widget>[
-          TextFormField(
-            validator: (value) {
-              if (value.isEmpty) {
-                return 'Please enter some text';
-              }
-              return null;
-            },
-          ),
-          Padding(
-            padding: const EdgeInsets.symmetric(vertical: 16.0),
-            child: ElevatedButton(
-              onPressed: () {
-                // Validate returns true if the form is valid, or false
-                // otherwise.
-                if (_formKey.currentState.validate()) {
-                  // If the form is valid, display a Snackbar.
-                  Scaffold.of(context)
-                      .showSnackBar(SnackBar(content: Text('Processing Data')));
-                }
-              },
-              child: Text('Submit'),
+      // IMPORTANT
+      child: LayoutBuilder(
+        builder: (context, constraints) => Column(
+          crossAxisAlignment: CrossAxisAlignment.start,
+          children: <Widget>[
+            TextFormField(
+              focusNode: _fieldFocusNode, // IMPORTANT
+              readOnly: _submitting, // IMPORTANT
+              onEditingComplete: _submit,
             ),
-          ),
-        ],
+            Padding(
+              padding: const EdgeInsets.symmetric(vertical: 16.0),
+              child: ElevatedButton(
+                onPressed: _submit,
+                child: Text('Submit'),
+              ),
+            ),
+          ],
+        ),
       ),
     );
   }
```
</details>

3. `flutter run`

4. enter some text in the text input

5. tap on "Submit"

6. tap the back arrow or the back android button

**Expected results:** It goes back to the previous (home) screen and the text input is focused.

**Actual results:** An error message in the screen:

<img height="720" src="https://user-images.githubusercontent.com/1031485/104137243-b49a9d00-539b-11eb-92a9-daea98fa7eb4.png">

And an error message and stack trace on the `flutter run` console:

```
I/flutter (23621): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
I/flutter (23621): The following assertion was thrown building UnmanagedRestorationScope:
I/flutter (23621): RenderBox.size accessed beyond the scope of resize, layout, or permitted parent access. RenderBox
I/flutter (23621): can always access its own size, otherwise, the only object that is allowed to read RenderBox.size is
I/flutter (23621): its parent, if they have said they will. It you hit this assert trying to access a child's size,
I/flutter (23621): pass "parentUsesSize: true" to that child's layout().
I/flutter (23621): 'package:flutter/src/rendering/box.dart':
I/flutter (23621): Failed assertion: line 1927 pos 13: 'debugDoingThisResize || debugDoingThisLayout ||
I/flutter (23621): _computingThisDryLayout ||
I/flutter (23621):               (RenderObject.debugActiveLayout == parent && _size._canBeUsedByParent)'
I/flutter (23621): 
I/flutter (23621): Either the assertion indicates an error in the framework itself, or we should provide substantially
I/flutter (23621): more information in this error message to help you determine and fix the underlying cause.
I/flutter (23621): In either case, please report this assertion by filing a bug on GitHub:
I/flutter (23621):   https://github.com/flutter/flutter/issues/new?template=2_bug.md
I/flutter (23621): 
I/flutter (23621): The relevant error-causing widget was:
I/flutter (23621):   TextFormField file:///home/luca/devel/flutter/bug11/lib/main.dart:86:13
I/flutter (23621): 
I/flutter (23621): When the exception was thrown, this was the stack:
[stack trace, see complete logs below]
```

<details>
  <summary>Logs</summary>

```console
$ flutter run -v
[  +56 ms] executing: [/home/luca/software/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[  +24 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] b0a22998593fc605c723dee8ff4d9315c32cfe2c
[        ] executing: [/home/luca/software/flutter/] git tag --points-at b0a22998593fc605c723dee8ff4d9315c32cfe2c
[  +13 ms] Exit code 0 from: git tag --points-at b0a22998593fc605c723dee8ff4d9315c32cfe2c
[        ] 1.25.0-8.2.pre
[  +30 ms] executing: [/home/luca/software/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[   +2 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] origin/beta
[        ] executing: [/home/luca/software/flutter/] git ls-remote --get-url origin
[   +2 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] https://github.com/flutter/flutter.git
[  +31 ms] executing: [/home/luca/software/flutter/] git rev-parse --abbrev-ref HEAD
[   +2 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] beta
[  +36 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +40 ms] executing: /home/luca/Android/Sdk/platform-tools/adb devices -l
[  +36 ms] List of devices attached
           LGH8701d9f7a4          device usb:1-2 product:lucye_global_com model:LG_H870 device:lucye transport_id:2
[   +5 ms] /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell getprop
[ +129 ms] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[   +2 ms] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[   +3 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[  +68 ms] Skipping pub get: version match.
[  +74 ms] Found plugin integration_test at /home/luca/software/flutter/packages/integration_test/
[ +101 ms] Found plugin integration_test at /home/luca/software/flutter/packages/integration_test/
[   +4 ms] Generating /home/luca/devel/flutter/bug11/android/app/src/main/java/io/flutter/plugins/GeneratedPluginRegistrant.java
[  +27 ms] ro.hardware = lucye
[        ] ro.build.characteristics = default
[  +29 ms] Initializing file store
[   +6 ms] Skipping target: gen_localizations
[   +3 ms] complete
[   +3 ms] Launching lib/main.dart on LG H870 in debug mode...
[   +3 ms] /home/luca/software/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/home/luca/software/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root
/home/luca/software/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --incremental --target=flutter --debugger-module-names
--experimental-emit-debug-metadata --output-dill /tmp/flutter_tools.BGJOOF/flutter_tool.IHOHBW/app.dill --packages
/home/luca/devel/flutter/bug11/.dart_tool/package_config.json -Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation
--filesystem-scheme org-dartlang-root --initialize-from-dill build/cache.dill.track.dill
[   +9 ms] executing: /home/luca/Android/Sdk/build-tools/30.0.2/aapt dump xmltree /home/luca/devel/flutter/bug11/build/app/outputs/flutter-apk/app.apk
AndroidManifest.xml
[   +3 ms] Exit code 0 from: /home/luca/Android/Sdk/build-tools/30.0.2/aapt dump xmltree
/home/luca/devel/flutter/bug11/build/app/outputs/flutter-apk/app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1e
               A: android:compileSdkVersionCodename(0x01010573)="11" (Raw: "11")
               A: package="com.example.bug11" (Raw: "com.example.bug11")
               A: platformBuildVersionCode=(type 0x10)0x1e
               A: platformBuildVersionName=(type 0x10)0xb
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1e
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: application (line=16)
                 A: android:label(0x01010001)="bug11" (Raw: "bug11")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=21)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.bug11.MainActivity" (Raw: "com.example.bug11.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=35)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=45)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw:
                     "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=49)
                     E: action (line=50)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=52)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=59)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +5 ms] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell -x logcat -v time -t 1
[   +6 ms] <- compile package:bug11/main.dart
[ +112 ms] --------- beginning of main
                    01-10 23:18:20.593 D/vendor.lge.hardware.thermal@1.0-impl(  625): Read value from /proc/lge_power/adc/vts, 326
[   +6 ms] executing: /home/luca/Android/Sdk/platform-tools/adb version
[   +3 ms] Android Debug Bridge version 1.0.41
           Version 30.0.5-6877874
           Installed as /home/luca/Android/Sdk/platform-tools/adb
[   +1 ms] executing: /home/luca/Android/Sdk/platform-tools/adb start-server
[   +3 ms] Building APK
[  +14 ms] Running Gradle task 'assembleDebug'...
[   +3 ms] Using gradle from /home/luca/devel/flutter/bug11/android/gradlew.
[        ] /home/luca/devel/flutter/bug11/android/gradlew mode: 33261 rwxr-xr-x.
[   +3 ms] executing: /home/luca/software/android-studio/jre/bin/java -version
[  +43 ms] Exit code 0 from: /home/luca/software/android-studio/jre/bin/java -version
[        ] openjdk version "1.8.0_242-release"
           OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
           OpenJDK 64-Bit Server VM (build 25.242-b3-6222593, mixed mode)
[   +1 ms] executing: [/home/luca/devel/flutter/bug11/android/] /home/luca/devel/flutter/bug11/android/gradlew -Pverbose=true
-Ptarget-platform=android-arm64 -Ptarget=/home/luca/devel/flutter/bug11/lib/main.dart -Ptrack-widget-creation=true -Pfilesystem-scheme=org-dartlang-root
assembleDebug
[ +475 ms] Welcome to Gradle 6.7!
[        ] Here are the highlights of this release:
[        ]  - File system watching is ready for production use
[        ]  - Declare the version of Java your build requires
[        ]  - Java 15 support
[        ] For more details see https://docs.gradle.org/6.7/release-notes.html
[+2388 ms] > Task :app:compileFlutterBuildDebug
[        ] [  +60 ms] executing: [/home/luca/software/flutter/] git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [  +28 ms] Exit code 0 from: git -c log.showSignature=false log -n 1 --pretty=format:%H
[        ] [        ] b0a22998593fc605c723dee8ff4d9315c32cfe2c
[        ] [        ] executing: [/home/luca/software/flutter/] git tag --points-at b0a22998593fc605c723dee8ff4d9315c32cfe2c
[        ] [  +12 ms] Exit code 0 from: git tag --points-at b0a22998593fc605c723dee8ff4d9315c32cfe2c
[        ] [        ] 1.25.0-8.2.pre
[        ] [  +34 ms] executing: [/home/luca/software/flutter/] git rev-parse --abbrev-ref --symbolic @{u}
[        ] [   +2 ms] Exit code 0 from: git rev-parse --abbrev-ref --symbolic @{u}
[        ] [        ] origin/beta
[        ] [        ] executing: [/home/luca/software/flutter/] git ls-remote --get-url origin
[        ] [   +2 ms] Exit code 0 from: git ls-remote --get-url origin
[        ] [        ] https://github.com/flutter/flutter.git
[        ] [  +30 ms] executing: [/home/luca/software/flutter/] git rev-parse --abbrev-ref HEAD
[        ] [   +2 ms] Exit code 0 from: git rev-parse --abbrev-ref HEAD
[        ] [        ] beta
[        ] [  +30 ms] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [   +1 ms] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [  +38 ms] Artifact Instance of 'MaterialFonts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'GradleWrapper' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidGenSnapshotArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'AndroidInternalBuildArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterWebSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterSdk' is not required, skipping update.
[        ] [        ] Artifact Instance of 'WindowsEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxEngineArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'LinuxFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'MacOSFuchsiaSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerSDKArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FlutterRunnerDebugSymbols' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'IosUsbArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'FontSubsetArtifacts' is not required, skipping update.
[        ] [        ] Artifact Instance of 'PubDependencies' is not required, skipping update.
[        ] [  +65 ms] Initializing file store
[        ] [   +5 ms] Done initializing file store
[        ] [  +24 ms] Skipping target: gen_localizations
[        ] [ +380 ms] kernel_snapshot: Starting due to {InvalidatedReason.inputChanged}
[        ] [   +8 ms] /home/luca/software/flutter/bin/cache/dart-sdk/bin/dart --disable-dart-dev
/home/luca/software/flutter/bin/cache/artifacts/engine/linux-x64/frontend_server.dart.snapshot --sdk-root
/home/luca/software/flutter/bin/cache/artifacts/engine/common/flutter_patched_sdk/ --target=flutter --no-print-incremental-dependencies
-Ddart.vm.profile=false -Ddart.vm.product=false --enable-asserts --track-widget-creation --no-link-platform --packages
/home/luca/devel/flutter/bug11/.dart_tool/package_config.json --output-dill
/home/luca/devel/flutter/bug11/.dart_tool/flutter_build/6207e47165b6408e77cf618632b71a05/app.dill --depfile
/home/luca/devel/flutter/bug11/.dart_tool/flutter_build/6207e47165b6408e77cf618632b71a05/kernel_snapshot.d package:bug11/main.dart
[+5489 ms] [+6591 ms] kernel_snapshot: Complete
[ +500 ms] [ +491 ms] debug_android_application: Starting due to {InvalidatedReason.inputChanged}
[ +200 ms] [ +161 ms] debug_android_application: Complete
[ +199 ms] [ +248 ms] Persisting file store
[        ] [   +5 ms] Done persisting file store
[        ] [   +4 ms] build succeeded.
[        ] [   +8 ms] "flutter assemble" took 8,046ms.
[        ] [   +2 ms] ensureAnalyticsSent: 0ms
[        ] [        ] Running shutdown hooks
[        ] [        ] Shutdown hooks complete
[        ] [        ] exiting with code 0
[ +200 ms] > Task :app:packLibsflutterBuildDebug UP-TO-DATE
[        ] > Task :app:preBuild UP-TO-DATE
[        ] > Task :app:preDebugBuild UP-TO-DATE
[        ] > Task :integration_test:preBuild UP-TO-DATE
[        ] > Task :integration_test:preDebugBuild UP-TO-DATE
[        ] > Task :integration_test:compileDebugAidl NO-SOURCE
[        ] > Task :app:compileDebugAidl NO-SOURCE
[        ] > Task :integration_test:packageDebugRenderscript NO-SOURCE
[        ] > Task :app:compileDebugRenderscript NO-SOURCE
[        ] > Task :app:generateDebugBuildConfig UP-TO-DATE
[        ] > Task :integration_test:writeDebugAarMetadata UP-TO-DATE
[        ] > Task :app:checkDebugAarMetadata UP-TO-DATE
[        ] > Task :app:cleanMergeDebugAssets
[        ] > Task :app:mergeDebugShaders UP-TO-DATE
[        ] > Task :app:compileDebugShaders NO-SOURCE
[        ] > Task :app:generateDebugAssets UP-TO-DATE
[        ] > Task :integration_test:mergeDebugShaders UP-TO-DATE
[        ] > Task :integration_test:compileDebugShaders NO-SOURCE
[        ] > Task :integration_test:generateDebugAssets UP-TO-DATE
[        ] > Task :integration_test:packageDebugAssets UP-TO-DATE
[        ] > Task :app:mergeDebugAssets
[ +195 ms] > Task :app:copyFlutterAssetsDebug
[        ] > Task :app:generateDebugResValues UP-TO-DATE
[        ] > Task :app:generateDebugResources UP-TO-DATE
[        ] > Task :integration_test:compileDebugRenderscript NO-SOURCE
[        ] > Task :integration_test:generateDebugResValues UP-TO-DATE
[        ] > Task :integration_test:generateDebugResources UP-TO-DATE
[        ] > Task :integration_test:packageDebugResources UP-TO-DATE
[        ] > Task :app:mergeDebugResources UP-TO-DATE
[        ] > Task :app:createDebugCompatibleScreenManifests UP-TO-DATE
[        ] > Task :app:extractDeepLinksDebug UP-TO-DATE
[        ] > Task :integration_test:extractDeepLinksDebug UP-TO-DATE
[        ] > Task :integration_test:processDebugManifest UP-TO-DATE
[        ] > Task :app:processDebugMainManifest UP-TO-DATE
[        ] > Task :app:processDebugManifest UP-TO-DATE
[        ] > Task :app:processDebugManifestForPackage UP-TO-DATE
[        ] > Task :integration_test:compileDebugLibraryResources UP-TO-DATE
[        ] > Task :integration_test:parseDebugLocalResources UP-TO-DATE
[        ] > Task :integration_test:generateDebugRFile UP-TO-DATE
[        ] > Task :app:processDebugResources UP-TO-DATE
[        ] > Task :integration_test:generateDebugBuildConfig UP-TO-DATE
[        ] > Task :integration_test:javaPreCompileDebug UP-TO-DATE
[        ] > Task :integration_test:compileDebugJavaWithJavac UP-TO-DATE
[        ] > Task :integration_test:bundleLibCompileToJarDebug UP-TO-DATE
[  +95 ms] > Task :app:compileDebugKotlin UP-TO-DATE
[        ] > Task :app:javaPreCompileDebug UP-TO-DATE
[        ] > Task :app:compileDebugJavaWithJavac UP-TO-DATE
[   +1 ms] > Task :app:compileDebugSources UP-TO-DATE
[        ] > Task :app:mergeDebugNativeDebugMetadata NO-SOURCE
[        ] > Task :app:processDebugJavaRes NO-SOURCE
[        ] > Task :integration_test:processDebugJavaRes NO-SOURCE
[        ] > Task :integration_test:bundleLibResDebug NO-SOURCE
[        ] > Task :app:mergeDebugJavaResource UP-TO-DATE
[        ] > Task :integration_test:bundleLibRuntimeToJarDebug UP-TO-DATE
[        ] > Task :app:checkDebugDuplicateClasses UP-TO-DATE
[        ] > Task :app:dexBuilderDebug UP-TO-DATE
[        ] > Task :app:desugarDebugFileDependencies UP-TO-DATE
[        ] > Task :app:mergeExtDexDebug UP-TO-DATE
[        ] > Task :app:mergeDexDebug UP-TO-DATE
[        ] > Task :app:mergeDebugJniLibFolders UP-TO-DATE
[        ] > Task :integration_test:mergeDebugJniLibFolders UP-TO-DATE
[        ] > Task :integration_test:mergeDebugNativeLibs NO-SOURCE
[        ] > Task :integration_test:stripDebugDebugSymbols NO-SOURCE
[        ] > Task :integration_test:copyDebugJniLibsProjectOnly UP-TO-DATE
[        ] > Task :app:mergeDebugNativeLibs UP-TO-DATE
[        ] > Task :app:stripDebugDebugSymbols UP-TO-DATE
[        ] > Task :app:validateSigningDebug UP-TO-DATE
[        ] > Task :integration_test:copyDebugJniLibsProjectAndLocalJars UP-TO-DATE
[        ] > Task :integration_test:extractDebugAnnotations UP-TO-DATE
[        ] > Task :integration_test:mergeDebugGeneratedProguardFiles UP-TO-DATE
[        ] > Task :integration_test:mergeDebugConsumerProguardFiles UP-TO-DATE
[        ] > Task :integration_test:prepareLintJarForPublish UP-TO-DATE
[        ] > Task :integration_test:mergeDebugJavaResource UP-TO-DATE
[        ] > Task :integration_test:syncDebugLibJars UP-TO-DATE
[  +94 ms] > Task :integration_test:bundleDebugAar UP-TO-DATE
[        ] > Task :integration_test:compileDebugSources UP-TO-DATE
[        ] > Task :integration_test:assembleDebug UP-TO-DATE
[ +399 ms] > Task :app:compressDebugAssets
[ +599 ms] > Task :app:packageDebug
[ +100 ms] > Task :app:assembleDebug
[  +99 ms] Deprecated Gradle features were used in this build, making it incompatible with Gradle 7.0.
[        ] Use '--warning-mode all' to show the individual deprecation warnings.
[        ] See https://docs.gradle.org/6.7/userguide/command_line_interface.html#sec:command_line_warnings
[        ] BUILD SUCCESSFUL in 10s
[        ] 57 actionable tasks: 7 executed, 50 up-to-date
[+1752 ms] Running Gradle task 'assembleDebug'... (completed in 12.8s)
[  +62 ms] calculateSha: LocalDirectory: '/home/luca/devel/flutter/bug11/build/app/outputs/flutter-apk'/app.apk
[  +18 ms] calculateSha: reading file took 17us
[ +728 ms] calculateSha: computing sha took 727us
[   +3 ms] ✓ Built build/app/outputs/flutter-apk/app-debug.apk.
[   +2 ms] executing: /home/luca/Android/Sdk/build-tools/30.0.2/aapt dump xmltree /home/luca/devel/flutter/bug11/build/app/outputs/flutter-apk/app.apk
AndroidManifest.xml
[   +3 ms] Exit code 0 from: /home/luca/Android/Sdk/build-tools/30.0.2/aapt dump xmltree
/home/luca/devel/flutter/bug11/build/app/outputs/flutter-apk/app.apk AndroidManifest.xml
[        ] N: android=http://schemas.android.com/apk/res/android
             E: manifest (line=2)
               A: android:versionCode(0x0101021b)=(type 0x10)0x1
               A: android:versionName(0x0101021c)="1.0.0" (Raw: "1.0.0")
               A: android:compileSdkVersion(0x01010572)=(type 0x10)0x1e
               A: android:compileSdkVersionCodename(0x01010573)="11" (Raw: "11")
               A: package="com.example.bug11" (Raw: "com.example.bug11")
               A: platformBuildVersionCode=(type 0x10)0x1e
               A: platformBuildVersionName=(type 0x10)0xb
               E: uses-sdk (line=7)
                 A: android:minSdkVersion(0x0101020c)=(type 0x10)0x10
                 A: android:targetSdkVersion(0x01010270)=(type 0x10)0x1e
               E: uses-permission (line=14)
                 A: android:name(0x01010003)="android.permission.INTERNET" (Raw: "android.permission.INTERNET")
               E: application (line=16)
                 A: android:label(0x01010001)="bug11" (Raw: "bug11")
                 A: android:icon(0x01010002)=@0x7f080000
                 A: android:debuggable(0x0101000f)=(type 0x12)0xffffffff
                 A: android:appComponentFactory(0x0101057a)="androidx.core.app.CoreComponentFactory" (Raw: "androidx.core.app.CoreComponentFactory")
                 E: activity (line=21)
                   A: android:theme(0x01010000)=@0x7f0a0000
                   A: android:name(0x01010003)="com.example.bug11.MainActivity" (Raw: "com.example.bug11.MainActivity")
                   A: android:launchMode(0x0101001d)=(type 0x10)0x1
                   A: android:configChanges(0x0101001f)=(type 0x11)0x40003fb4
                   A: android:windowSoftInputMode(0x0101022b)=(type 0x11)0x10
                   A: android:hardwareAccelerated(0x010102d3)=(type 0x12)0xffffffff
                   E: meta-data (line=35)
                     A: android:name(0x01010003)="io.flutter.embedding.android.NormalTheme" (Raw: "io.flutter.embedding.android.NormalTheme")
                     A: android:resource(0x01010025)=@0x7f0a0001
                   E: meta-data (line=45)
                     A: android:name(0x01010003)="io.flutter.embedding.android.SplashScreenDrawable" (Raw:
                     "io.flutter.embedding.android.SplashScreenDrawable")
                     A: android:resource(0x01010025)=@0x7f040000
                   E: intent-filter (line=49)
                     E: action (line=50)
                       A: android:name(0x01010003)="android.intent.action.MAIN" (Raw: "android.intent.action.MAIN")
                     E: category (line=52)
                       A: android:name(0x01010003)="android.intent.category.LAUNCHER" (Raw: "android.intent.category.LAUNCHER")
                 E: meta-data (line=59)
                   A: android:name(0x01010003)="flutterEmbedding" (Raw: "flutterEmbedding")
                   A: android:value(0x01010024)=(type 0x10)0x2
[   +1 ms] Stopping app 'app.apk' on LG H870.
[        ] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell am force-stop com.example.bug11
[ +127 ms] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell pm list packages com.example.bug11
[  +58 ms] package:com.example.bug11
[   +5 ms] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell cat /data/local/tmp/sky.com.example.bug11.sha1
[  +36 ms] 216680ceb1e29516f0cf1eb11097dfbbad89e66b
[        ] Installing APK.
[   +1 ms] Installing build/app/outputs/flutter-apk/app.apk...
[        ] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 install -t -r
/home/luca/devel/flutter/bug11/build/app/outputs/flutter-apk/app.apk
[+11626 ms] Performing Streamed Install
                     Success
[        ] Installing build/app/outputs/flutter-apk/app.apk... (completed in 11.6s)
[   +2 ms] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell echo -n 7d7aab404399786d6e891f224698e3b08c91ced7 >
/data/local/tmp/sky.com.example.bug11.sha1
[  +60 ms] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell -x logcat -v time -t 1
[  +92 ms] --------- beginning of system
           01-10 23:18:46.634 I/MaxAspectRatioPackages( 4283): packageName : com.example.bug11 has been updated, but set as default
[  +12 ms] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 shell am start -a android.intent.action.RUN -f 0x20000000 --ez
enable-background-compilation true --ez enable-dart-profiling true --ez enable-checked-mode true --ez verify-entry-points true
com.example.bug11/com.example.bug11.MainActivity
[ +481 ms] Starting: Intent { act=android.intent.action.RUN flg=0x20000000 cmp=com.example.bug11/.MainActivity (has extras) }
[        ] Waiting for observatory port to be available...
[+3155 ms] Observatory URL on device: http://127.0.0.1:57218/TLnEkPGAHBA=/
[   +3 ms] executing: /home/luca/Android/Sdk/platform-tools/adb -s LGH8701d9f7a4 forward tcp:0 tcp:57218
[  +13 ms] 43561
[   +1 ms] Forwarded host port 43561 to device port 57218 for Observatory
[  +16 ms] Caching compiled dill
[  +38 ms] Connecting to service protocol: http://127.0.0.1:43561/TLnEkPGAHBA=/
[   +2 ms] Launching a Dart Developer Service (DDS) instance at http://127.0.0.1:0, connecting to VM service at http://127.0.0.1:43561/TLnEkPGAHBA=/.
[+1063 ms] DDS is listening at http://127.0.0.1:34901/BZo3WNjEF2M=/.
[  +53 ms] Successfully connected to service protocol: http://127.0.0.1:43561/TLnEkPGAHBA=/
[  +38 ms] DevFS: Creating new filesystem on the device (null)
[ +104 ms] DevFS: Created new filesystem on the device (file:///data/user/0/com.example.bug11/code_cache/bug11AOBHXY/bug11/)
[   +5 ms] Updating assets
[  +99 ms] Syncing files to device LG H870...
[   +1 ms] <- reset
[        ] Compiling dart to kernel with 0 updated files
[   +1 ms] <- recompile package:bug11/main.dart 68b80b6a-e675-4805-8565-f66ef92f664c
[        ] <- 68b80b6a-e675-4805-8565-f66ef92f664c
[  +64 ms] Updating files.
[        ] DevFS: Sync finished
[        ] Syncing files to device LG H870... (completed in 68ms)
[        ] Synced 0.0MB.
[        ] <- accept
[  +49 ms] Connected to _flutterView/0x74f3032020.
[   +1 ms] Flutter run key commands.
[   +2 ms] r Hot reload. 🔥🔥🔥
[   +1 ms] R Hot restart.
[        ] h Repeat this help message.
[        ] d Detach (terminate "flutter run" but leave application running).
[        ] c Clear the screen
[        ] q Quit (terminate the application on the device).
[        ] An Observatory debugger and profiler on LG H870 is available at: http://127.0.0.1:34901/BZo3WNjEF2M=/
[        ] Running with unsound null safety
[        ] For more information see https://dart.dev/null-safety/unsound-null-safety
[+238328 ms] I/ViewRootImpl(29387): ViewRoot's Touch Event : ACTION_DOWN
[  +63 ms] I/ViewRootImpl(29387): ViewRoot's Touch Event : ACTION_UP
[ +480 ms] I/ViewRootImpl(29387): ViewRoot's Touch Event : ACTION_DOWN
[  +67 ms] I/ViewRootImpl(29387): ViewRoot's Touch Event : ACTION_UP
[ +274 ms] W/IInputConnectionWrapper(29387): getTextBeforeCursor on inactive InputConnection
[   +3 ms] W/IInputConnectionWrapper(29387): getSelectedText on inactive InputConnection
[   +4 ms] W/IInputConnectionWrapper(29387): getTextAfterCursor on inactive InputConnection
[  +28 ms] W/IInputConnectionWrapper(29387): beginBatchEdit on inactive InputConnection
[        ] W/IInputConnectionWrapper(29387): endBatchEdit on inactive InputConnection
[+1898 ms] I/ViewRootImpl(29387): ViewRoot's Touch Event : ACTION_DOWN
[  +59 ms] I/ViewRootImpl(29387): ViewRoot's Touch Event : ACTION_UP
[+1832 ms] I/ViewRootImpl(29387): ViewRoot's KeyEvent { action=ACTION_DOWN, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x48, repeatCount=0,
eventTime=81566302, downTime=81566302, deviceId=-1, source=0x101 } to DecorView@5ca0b75[MainActivity]
[  +41 ms] I/ViewRootImpl(29387): ViewRoot's KeyEvent { action=ACTION_UP, keyCode=KEYCODE_BACK, scanCode=0, metaState=0, flags=0x48, repeatCount=0,
eventTime=81566360, downTime=81566302, deviceId=-1, source=0x101 } to DecorView@5ca0b75[MainActivity]
[ +595 ms] I/flutter (29387): ══╡ EXCEPTION CAUGHT BY WIDGETS LIBRARY ╞═══════════════════════════════════════════════════════════
[        ] I/flutter (29387): The following assertion was thrown building UnmanagedRestorationScope:
[        ] I/flutter (29387): RenderBox.size accessed beyond the scope of resize, layout, or permitted parent access. RenderBox
[        ] I/flutter (29387): can always access its own size, otherwise, the only object that is allowed to read RenderBox.size is
[        ] I/flutter (29387): its parent, if they have said they will. It you hit this assert trying to access a child's size,
[        ] I/flutter (29387): pass "parentUsesSize: true" to that child's layout().
[        ] I/flutter (29387): 'package:flutter/src/rendering/box.dart':
[        ] I/flutter (29387): Failed assertion: line 1927 pos 13: 'debugDoingThisResize || debugDoingThisLayout ||
[        ] I/flutter (29387): _computingThisDryLayout ||
[        ] I/flutter (29387):               (RenderObject.debugActiveLayout == parent && _size._canBeUsedByParent)'
[        ] I/flutter (29387): 
[        ] I/flutter (29387): Either the assertion indicates an error in the framework itself, or we should provide substantially
[        ] I/flutter (29387): more information in this error message to help you determine and fix the underlying cause.
[        ] I/flutter (29387): In either case, please report this assertion by filing a bug on GitHub:
[        ] I/flutter (29387):   https://github.com/flutter/flutter/issues/new?template=2_bug.md
[        ] I/flutter (29387): 
[        ] I/flutter (29387): The relevant error-causing widget was:
[        ] I/flutter (29387):   TextFormField file:///home/luca/devel/flutter/bug11/lib/main.dart:84:13
[        ] I/flutter (29387): 
[        ] I/flutter (29387): When the exception was thrown, this was the stack:
[        ] I/flutter (29387): #2      RenderBox.size.<anonymous closure> (package:flutter/src/rendering/box.dart:1927:13)
[        ] I/flutter (29387): #3      RenderBox.size (package:flutter/src/rendering/box.dart:1940:6)
[        ] I/flutter (29387): #4      EditableTextState._updateSizeAndTransform (package:flutter/src/widgets/editable_text.dart:2362:40)
[        ] I/flutter (29387): #5      EditableTextState._openInputConnection (package:flutter/src/widgets/editable_text.dart:1981:7)
[        ] I/flutter (29387): #6      EditableTextState.didUpdateWidget (package:flutter/src/widgets/editable_text.dart:1576:9)
[        ] I/flutter (29387): #7      StatefulElement.update (package:flutter/src/widgets/framework.dart:4872:57)
[        ] I/flutter (29387): #8      Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #9      ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #10     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #11     ProxyElement.update (package:flutter/src/widgets/framework.dart:5043:5)
[        ] I/flutter (29387): #12     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #13     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6181:14)
[        ] I/flutter (29387): #14     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #15     _DecorationElement._updateChild (package:flutter/src/material/input_decorator.dart:1597:31)
[        ] I/flutter (29387): #16     _DecorationElement.update (package:flutter/src/material/input_decorator.dart:1611:5)
[  +36 ms] I/flutter (29387): #17     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #18     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #19     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #20     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #21     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #22     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #23     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #24     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #25     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #26     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #27     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #28     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6181:14)
[        ] I/flutter (29387): #29     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #30     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #31     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #32     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #33     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #34     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #35     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #36     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #37     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #38     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #39     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #40     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6181:14)
[        ] I/flutter (29387): #41     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #42     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #43     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #44     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #45     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #46     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #47     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6181:14)
[        ] I/flutter (29387): #48     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #49     SingleChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6181:14)
[        ] I/flutter (29387): #50     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #51     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #52     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #53     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #54     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #55     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #56     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #57     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #58     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #59     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #60     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #61     ComponentElement.performRebuild (package:flutter/src/widgets/framework.dart:4710:16)
[        ] I/flutter (29387): #62     StatefulElement.performRebuild (package:flutter/src/widgets/framework.dart:4857:11)
[        ] I/flutter (29387): #63     Element.rebuild (package:flutter/src/widgets/framework.dart:4379:5)
[        ] I/flutter (29387): #64     StatefulElement.update (package:flutter/src/widgets/framework.dart:4889:5)
[        ] I/flutter (29387): #65     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #66     RenderObjectElement.updateChildren (package:flutter/src/widgets/framework.dart:5703:32)
[        ] I/flutter (29387): #67     MultiChildRenderObjectElement.update (package:flutter/src/widgets/framework.dart:6303:17)
[        ] I/flutter (29387): #68     Element.updateChild (package:flutter/src/widgets/framework.dart:3377:15)
[        ] I/flutter (29387): #69     _LayoutBuilderElement._layout.<anonymous closure> (package:flutter/src/widgets/layout_builder.dart:136:18)
[        ] I/flutter (29387): #70     BuildOwner.buildScope (package:flutter/src/widgets/framework.dart:2731:19)
[        ] I/flutter (29387): #71     _LayoutBuilderElement._layout (package:flutter/src/widgets/layout_builder.dart:118:12)
[        ] I/flutter (29387): #72     RenderObject.invokeLayoutCallback.<anonymous closure> (package:flutter/src/rendering/object.dart:1887:59)
[        ] I/flutter (29387): #73     PipelineOwner._enableMutationsToDirtySubtrees (package:flutter/src/rendering/object.dart:915:15)
[        ] I/flutter (29387): #74     RenderObject.invokeLayoutCallback (package:flutter/src/rendering/object.dart:1887:14)
[        ] I/flutter (29387): #75     RenderConstrainedLayoutBuilder.rebuildIfNecessary (package:flutter/src/widgets/layout_builder.dart:225:7)
[        ] I/flutter (29387): #76     _RenderLayoutBuilder.performLayout (package:flutter/src/widgets/layout_builder.dart:360:5)
[        ] I/flutter (29387): #77     RenderObject.layout (package:flutter/src/rendering/object.dart:1777:7)
[        ] I/flutter (29387): #78     MultiChildLayoutDelegate.layoutChild (package:flutter/src/rendering/custom_layout.dart:171:12)
[        ] I/flutter (29387): #79     _ScaffoldLayout.performLayout (package:flutter/src/material/scaffold.dart:911:7)
[        ] I/flutter (29387): #80     MultiChildLayoutDelegate._callPerformLayout (package:flutter/src/rendering/custom_layout.dart:243:7)
[        ] I/flutter (29387): #81     RenderCustomMultiChildLayoutBox.performLayout (package:flutter/src/rendering/custom_layout.dart:407:14)
[        ] I/flutter (29387): #82     RenderObject._layoutWithoutResize (package:flutter/src/rendering/object.dart:1634:7)
[        ] I/flutter (29387): #83     PipelineOwner.flushLayout (package:flutter/src/rendering/object.dart:884:18)
[        ] I/flutter (29387): #84     RendererBinding.drawFrame (package:flutter/src/rendering/binding.dart:454:19)
[        ] I/flutter (29387): #85     WidgetsBinding.drawFrame (package:flutter/src/widgets/binding.dart:900:13)
[        ] I/flutter (29387): #86     RendererBinding._handlePersistentFrameCallback (package:flutter/src/rendering/binding.dart:320:5)
[        ] I/flutter (29387): #87     SchedulerBinding._invokeFrameCallback (package:flutter/src/scheduler/binding.dart:1119:15)
[        ] I/flutter (29387): #88     SchedulerBinding.handleDrawFrame (package:flutter/src/scheduler/binding.dart:1057:9)
[        ] I/flutter (29387): #89     SchedulerBinding._handleDrawFrame (package:flutter/src/scheduler/binding.dart:973:5)
[        ] I/flutter (29387): #93     _invoke (dart:ui/hooks.dart:157:10)
[        ] I/flutter (29387): #94     PlatformDispatcher._drawFrame (dart:ui/platform_dispatcher.dart:253:5)
[        ] I/flutter (29387): #95     _drawFrame (dart:ui/hooks.dart:120:31)
[        ] I/flutter (29387): (elided 5 frames from class _AssertionError and dart:async)
[        ] I/flutter (29387): 
[        ] I/flutter (29387): ════════════════════════════════════════════════════════════════════════════════════════════════════
[  +38 ms] I/flutter (29387): Another exception was thrown: A RenderFlex overflowed by 99516 pixels on the bottom.
[ +126 ms] I/flutter (29387): Another exception was thrown: A GlobalKey was used multiple times inside one widget's child list.
```

```console
$ flutter analyze 
Analyzing bug11...                                                      
No issues found! (ran in 1.7s)
```

```console
$ flutter doctor -v
[✓] Flutter (Channel beta, 1.25.0-8.2.pre, on Linux, locale en_IE.UTF-8)
    • Flutter version 1.25.0-8.2.pre at /home/luca/software/flutter
    • Framework revision b0a2299859 (5 days ago), 2021-01-05 12:34:13 -0800
    • Engine revision 92ae191c17
    • Dart version 2.12.0 (build 2.12.0-133.2.beta)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /home/luca/Android/Sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /home/luca/software/android-studio/jre/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)
    • All Android licenses accepted.

[✓] Chrome - develop for the web
    • CHROME_EXECUTABLE = chromium

[✓] Android Studio
    • Android Studio at /home/luca/software/android-studio
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • android-studio-dir = /home/luca/software/android-studio
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6222593)

[✓] Connected device (2 available)
    • LG H870 (mobile) • LGH8701d9f7a4 • android-arm64  • Android 9 (API 28)
    • Chrome (web)     • chrome        • web-javascript • Chromium 83.0.4103.116 built on Debian bullseye/sid, running on Debian bullseye/sid

• No issues found!
```

</details>

**Comment by llucax** [comment]
Note: Moving the `Navigator.push()` between the `setState()` and the `_fieldFocusNode.requestFocus();` (or after both, i.e. to the end of the method) makes the error go away too.

**Comment by llucax** [comment]
For now I'm using that as a workaround, but I'm not sure if I'm hitting some Flutter bug or if I'm doing something wrong and by moving the `Navigator.push` I'm just going around some Flutter sanity check and it will blow up in my face in the future, so it would be very interesting if anyone can explain which is it :)

**Comment by pedromassangocode** [comment]
I was able to reproduce the issue on latest `master` channel.


<details>
<summary>flutter doctor -v</summary>

```bash
[✓] Flutter (Channel master, 1.26.0-2.0.pre.275, on Mac OS X 10.15.7 19H2 darwin-x64, locale en)
    • Flutter version 1.26.0-2.0.pre.275 at /Users/pedromassango/Code/flutter_master
    • Framework revision d9044a8a61 (2 days ago), 2021-01-09 04:04:04 -0500
    • Engine revision caf6a8191f
    • Dart version 2.12.0 (build 2.12.0-204.0.dev)

[✓] Android toolchain - develop for Android devices (Android SDK version 30.0.2)
    • Android SDK at /Users/pedromassango/Library/Android/sdk
    • Platform android-30, build-tools 30.0.2
    • Java binary at: /Applications/Android Studio.app/Contents/jre/jdk/Contents/Home/bin/java
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)
    • All Android licenses accepted.

[!] Xcode - develop for iOS and macOS (Xcode 12.1)
    • Xcode at /Applications/Xcode.app/Contents/Developer
    • Xcode 12.1, Build version 12A7403
    ! CocoaPods 1.9.3 out of date (1.10.0 is recommended).
        CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
        Without CocoaPods, plugins will not work on iOS or macOS.
        For more info, see https://flutter.dev/platform-plugins
      To upgrade see https://guides.cocoapods.org/using/getting-started.html#installation for instructions.

[✓] Chrome - develop for the web
    • Chrome at /Applications/Google Chrome.app/Contents/MacOS/Google Chrome

[✓] Android Studio (version 4.1)
    • Android Studio at /Applications/Android Studio.app/Contents
    • Flutter plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/9212-flutter
    • Dart plugin can be installed from:
      🔨 https://plugins.jetbrains.com/plugin/6351-dart
    • Java version OpenJDK Runtime Environment (build 1.8.0_242-release-1644-b3-6915495)

[✓] IntelliJ IDEA Community Edition (version 2020.3.1)
    • IntelliJ at /Applications/IntelliJ IDEA CE.app
    • Flutter plugin version 52.1.5
    • Dart plugin version 203.6912

[✓] VS Code (version 1.52.1)
    • VS Code at /Applications/Visual Studio Code.app/Contents
    • Flutter extension version 3.18.1

[✓] Connected device (4 available)
    • Redmi 5 Plus (mobile)      • 0258ff700005  • android-arm64  • Android 8.1.0 (API 27)
    • sdk gphone x86 64 (mobile) • emulator-5554 • android-x64    • Android 11 (API 30) (emulator)
    • macOS (desktop)            • macos         • darwin-x64     • Mac OS X 10.15.7 19H2 darwin-x64
    • Chrome (web)               • chrome        • web-javascript • Google Chrome 87.0.4280.88

! Doctor found issues in 1 category.
```

</details>

**Comment by llucax** [comment]
In my real app I'm running out of workarounds, now no matter how I order the `Navigator.push()` call, I hit this, so it would be appreciated if some *expert* can have a look at the code to see if there is anything wrong with it (bad usage) or if it is a proper framework bug.

**Comment by HansMuller** [comment]
CC @chunhtai @goderbauer

**Comment by llucax** [comment]
I just tried with the stable channel (I'm using beta because I'm targeting the web too) and I get the same error (after a `flutter channel stable && flutter upgrade && flutter clean && flutter run`). Only the assertion location changes:
```
I/flutter ( 3206): 'package:flutter/src/rendering/box.dart':
I/flutter ( 3206): Failed assertion: line 1792 pos 13: 'debugDoingThisResize || debugDoingThisLayout ||
I/flutter ( 3206):               (RenderObject.debugActiveLayout == parent && _size._canBeUsedByParent)'
```

**Comment by chunhtai** [comment]
the globalkey error is a result of the first error
```
I/flutter ( 3206): Failed assertion: line 1792 pos 13: 'debugDoingThisResize || debugDoingThisLayout ||
I/flutter ( 3206):               (RenderObject.debugActiveLayout == parent && _size._canBeUsedByParent)'
```
it is weird the screen prints the last error instead of the first though

**Comment by llucax** [comment]
I see, should I update the issue's title?

I still don't know if I'm doing something wrong or if this is a framework bug. I can't really try the suggestion ("It you hit this assert trying to access a child's size, pass "parentUsesSize: true" to that child's layout().") because I'm not using `RenderBox` or manipulating sizes anywhere directly.

**Comment by llucax** [comment]
Actually the title is at least descriptive of one of the issues that seem to be present here. The screen is showing the *wrong* error. And then there is the underlying problem causing the issue in the first place and the suggestion that if this is not a framework bug, a better error should be provided so users can fix their code.

Not sure if this should be split in two different GitHub issues.

**Comment by chunhtai** [comment]
at a glance this looks like a framework bug, the screen prints the wrong error should be split out to a separate issue.

**Comment by llucax** [comment]
If anyone has any suggestions on how to work around the issue that is less disruptive than removing the `LayoutBuilder` would be very appreciated :)

**Comment by llucax** [comment]
Awesome! Is there any way to know when is this fix expected to land in the Flutter beta channel?

**Comment by llucax** [comment]
Anywhere where I can get this information? I'm really blocked by this bug, I hope it can be ported at least to the beta channel soon.

**Comment by xu-baolin** [comment]
I think it will take a couple of weeks. and it doesn't matter on release version,I mean the assertion will not check on release version app.

**Comment by llucax** [comment]
Thanks @xu-baolin, that is at least something, but it still breaks all my CI and testing/debugging environment, building all and everything with `--release` is not a practical solution for me :(

**Comment by llucax** [comment]
The fix is in the new beta released today. Thanks!

**Comment by pcsosinski** [comment]
no CP required
